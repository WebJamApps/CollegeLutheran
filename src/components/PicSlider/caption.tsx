import React from 'react';

const Caption = (props: { caption?: (JSX.Element|string); }):JSX.Element => {
  const { caption } = props;
  return (
    <div
      className="slider-caption"
      style={{
        textAlign: 'center',
        fontWeight: 'bold',
        padding: 0,
        backgroundColor: 'silver',
        marginBottom: 0,
        marginTop: 0,
      }}
    >
      {' '}
      {caption}
      {' '}
    </div>
  );
};

Caption.defaultProps = {
  caption: '',
};

export default Caption;
