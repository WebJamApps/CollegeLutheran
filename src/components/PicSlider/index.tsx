import React, { Component } from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Caption from './caption';
import { Ibook } from '../../redux/mapStoreToProps';

type SettingsType = {
  autoplay: boolean;
  autoplaySpeed: number;
  infinite: boolean;
  speed: number;
  slidesToShow: number;
  slidesToScroll: number;
  arrows: boolean;
  fade: boolean;
};

export interface PicSliderProps {
  settings?: SettingsType;
  data: Ibook[];
}
class PicSlider extends Component<PicSliderProps> {
  static defaultProps: { data: [{ url: ''; title: ''; _id: 0 }]; };

  settings: SettingsType;

  constructor(props: Readonly<PicSliderProps>) {
    super(props);
    this.settings = {
      autoplay: true,
      autoplaySpeed: 3000,
      infinite: true,
      speed: 5000,
      slidesToShow: 1,
      slidesToScroll: 1,
      arrows: false,
      fade: true,
    };
  }

  /* eslint-disable react/jsx-props-no-spreading */
  render(): JSX.Element {
    const { data } = this.props;
    return (
      <div>
        <Slider {...this.settings}>
          {
            data.map((d: Ibook) => (
              <div key={d._id}>
                <img className="slide-show" src={d.url} alt={d.title} />
                {d.comments === 'showCaption' ? <Caption caption={d.title} /> : null}
              </div>
            ))
          }
        </Slider>
      </div>
    );
  }
}

export default PicSlider;
