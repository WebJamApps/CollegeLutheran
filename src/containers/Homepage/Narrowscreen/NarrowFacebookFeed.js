import React from 'react';

const FacebookFeed = () => (
  <div className="notWidescreen" style={{ maxWidth: '320px', margin: 'auto' }}>
    <p style={{
      textAlign: 'center', fontSize: '10pt', marginTop: 0, marginBottom: 0, paddingBottom: '2px',
    }}
    >
      <a style={{ fontSize: '10pt' }} href="/calendar">
        <i>View Full Calendar</i>
      </a>
    </p>
    <div style={{ maxWidth: '300px', margin: 'auto', marginBottom: 0 }}>
      <iframe
        title="clc-calendar"/* eslint-disable-next-line max-len */
        src="https://calendar.google.com/calendar/embed?title=College%20Lutheran%20Events%20Calendar&amp;showNav=0&amp;showDate=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;mode=AGENDA&amp;height=500&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=ra8um2u26p59a2t92lfooq4els%40group.calendar.google.com&amp;color=%23060D5E&amp;ctz=America%2FNew_York"
        style={{ borderWidth: 0 }}
        width="300"
        height="300"
        frameBorder="0"
        scrolling="no"
      />
    </div>
    <hr />
    <p style={{
      textAlign: 'center', fontSize: '10pt', marginTop: '10px', marginBottom: 0, paddingBottom: '4px',
    }}
    >
      <i>
        Like Us On
        {' '}
        <a style={{ fontSize: '10pt' }} href="https://www.facebook.com/CollegeLutheranChurch/">Facebook</a>
      </i>
    </p>
    <div style={{ maxWidth: '300px', margin: 'auto', marginBottom: 0 }}>
      <iframe
        title="clc-facebook"/* eslint-disable-next-line max-len */
        src="//www.facebook.com/plugins/likebox.php?href=https%3A%2F%2Fwww.facebook.com%2FCollegeLutheranChurch&width=300&height=500&colorscheme=light&show_faces=false&header=true&stream=true&show_border=false"
        width="300"
        height="300"
        style={{ border: 'none', overflow: 'hidden' }}
        scrolling="no"
        frameBorder="0"
        allow="encrypted-media"
      />
    </div>
    <p>{' '}</p>
  </div>
);

export default FacebookFeed;
