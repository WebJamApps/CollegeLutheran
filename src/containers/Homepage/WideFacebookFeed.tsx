import React from 'react';

interface IWideFBFeed {
  width?: number;
}

const WideFacebookFeed = ({ width }: IWideFBFeed): JSX.Element => {
  let mb = '32px', mt = '0px';
  if (width && width < 1092) { mb = '0px'; mt = '32px'; }
  return (
    <div className="row" style={{ marginLeft: '10px' }}>
      <div className="col" style={{ maxWidth: '600px', margin: 'auto', paddingRight: '10px' }}>
        <p style={{ textAlign: 'center', fontSize: '10pt', marginTop: '0px' }}>
          <a style={{ fontSize: '10pt' }} href="/calendar">
            <i>View Full Calendar</i>
          </a>
        </p>
        <iframe // eslint-disable-next-line max-len
          src="https://calendar.google.com/calendar/embed?title=College%20Lutheran%20Events%20Calendar&amp;showNav=0&amp;showDate=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;mode=AGENDA&amp;height=500&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=ra8um2u26p59a2t92lfooq4els%40group.calendar.google.com&amp;color=%23060D5E&amp;ctz=America%2FNew_York"
          style={{ borderWidth: 0 }}
          width="550"
          height="500"
          frameBorder="0"
          scrolling="no"
          title="google-calendar"
        />
      </div>
      <div
        className="col"
        style={{
          width: '600px', margin: 'auto', paddingRight: '10px', paddingLeft: '10px', textAlign: 'center',
        }}
      >
        <p
          id="wideFacebook"
          style={{
            textAlign: 'center', fontSize: '10pt', marginBottom: mb, marginTop: mt,
          }}
        >
          <i>
            Like Us On
            {' '}
            <a style={{ fontSize: '10pt' }} href="https://www.facebook.com/CollegeLutheranChurch/">Facebook</a>
          </i>
        </p>
        <iframe
          className="widescreenHomepage"
          // eslint-disable-next-line max-len
          src="//www.facebook.com/plugins/likebox.php?href=https%3A%2F%2Fwww.facebook.com%2FCollegeLutheranChurch&width=500&height=485&colorscheme=light&show_faces=false&header=true&stream=true&show_border=false"
          width="500"
          height="485"
          style={{ border: 'none', overflow: 'hidden', marginLeft: '-14px' }}
          scrolling="none"
          allow="encrypted-media"
          title="facebook ticker"
        />
      </div>
    </div>
  );
};
WideFacebookFeed.defaultProps = { width: 1004 };

export default WideFacebookFeed;
