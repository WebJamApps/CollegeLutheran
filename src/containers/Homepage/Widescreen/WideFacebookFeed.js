import React from 'react';

const WideFacebookFeed = () => (
  <div className="row" style={{ marginLeft: '10px' }}>
    <div className="col" style={{ maxWidth: '600px', margin: 'auto', paddingRight: '10px' }}>
      <p style={{ textAlign: 'center', fontSize: '10pt', marginTop: '0px' }}>
        <a style={{ fontSize: '10pt' }} href="/calendar">
          <i>View Full Calendar</i>
        </a>
      </p>
      <iframe // eslint-disable-next-line max-len
        src="https://calendar.google.com/calendar/embed?title=College%20Lutheran%20Events%20Calendar&amp;showNav=0&amp;showDate=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;mode=AGENDA&amp;height=500&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=ra8um2u26p59a2t92lfooq4els%40group.calendar.google.com&amp;color=%23060D5E&amp;ctz=America%2FNew_York"
        style={{ borderWidth: 0 }}
        width="550"
        height="500"
        frameBorder="0"
        scrolling="no"
        title="google-calendar"
      />
    </div>
    <div
      className="col"
      style={{
        width: '600px', margin: 'auto', paddingRight: '10px', paddingLeft: '10px', textAlign: 'center',
      }}
    >
      <p style={{
        textAlign: 'center', fontSize: '10pt', marginTop: '0px', marginBottom: '32px',
      }}
      >
        <i>
        Like Us On
          {' '}
          <a style={{ fontSize: '10pt' }} href="https://www.facebook.com/CollegeLutheranChurch/">Facebook</a>
        </i>
      </p>
      <iframe
        className="widescreenHomepage"
      // eslint-disable-next-line max-len
        src="//www.facebook.com/plugins/likebox.php?href=https%3A%2F%2Fwww.facebook.com%2FCollegeLutheranChurch&width=500&height=485&colorscheme=light&show_faces=false&header=true&stream=true&show_border=false"
        width="500"
        height="485"
        style={{ border: 'none', overflow: 'hidden', marginLeft: '-14px' }}
        scrolling="none"
        allow="encrypted-media"
        title="facebook ticker"
      />
    </div>
  </div>
);

export default WideFacebookFeed;
