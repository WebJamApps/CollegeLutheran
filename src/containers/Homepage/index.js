import React, { Component } from 'react';
import ReactResizeDetector from 'react-resize-detector';
import WideAboutUs from './Widescreen/WideAbout';
import WideCurrentProjects from './Widescreen/WideCurrentProjects';
import NarrowAboutUs from './Narrowscreen/NarrowAbout';
import NarrowCurrentProjects from './Narrowscreen/NarrowCurrentProjects';
import FacebookFeed from './Narrowscreen/NarrowFacebookFeed';
import Inquiry from '../../components/inquiry';

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.parentRef = React.createRef();
    this.onResize = this.onResize.bind(this);
    this.state = { width: 100, nodeEnv: process.env.NODE_ENV };
  }

  onResize(width) {
    this.setState({ width });
  }

  render() {
    const { width, nodeEnv } = this.state;
    return (
      <div>
        {width >= 1004
          ? (
            <div className="page-content">
              <WideAboutUs />
              <hr />
              <WideCurrentProjects />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
            </div>
          )
          : (
            <div className="page-content">
              <NarrowAboutUs />
              <hr />
              <NarrowCurrentProjects />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
              <FacebookFeed />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
              {nodeEnv !== 'production' ? <Inquiry /> : null}
            </div>
          )}
        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} targetDomEl={this.parentRef.current} />
      </div>
    );
  }
}
