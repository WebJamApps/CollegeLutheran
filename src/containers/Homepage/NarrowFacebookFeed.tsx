import React from 'react';
import PicSlider from '../../components/PicSlider';
import { Ibook } from '../../redux/mapStoreToProps';

interface IallPics {
  allPics?: Ibook[]
}

const FacebookFeed = ({ allPics }: IallPics): JSX.Element => (
  <div className="notWidescreen" style={{ maxWidth: '320px', margin: 'auto' }}>
    <p style={{
      textAlign: 'center', fontSize: '10pt', marginTop: 0, marginBottom: 0, paddingBottom: '2px',
    }}
    >
      <a style={{ fontSize: '10pt' }} href="/calendar">
        <i>View Full Calendar</i>
      </a>
    </p>
    <div style={{ maxWidth: '300px', margin: 'auto', marginBottom: 0 }}>
      <iframe
        title="clc-calendar"/* eslint-disable-next-line max-len */
        src="https://calendar.google.com/calendar/embed?title=College%20Lutheran%20Events%20Calendar&amp;showNav=0&amp;showDate=0&amp;showPrint=0&amp;showTabs=0&amp;showCalendars=0&amp;mode=AGENDA&amp;height=500&amp;wkst=1&amp;bgcolor=%23FFFFFF&amp;src=ra8um2u26p59a2t92lfooq4els%40group.calendar.google.com&amp;color=%23060D5E&amp;ctz=America%2FNew_York"
        style={{ borderWidth: 0 }}
        width="300"
        height="300"
        frameBorder="0"
        scrolling="no"
      />
    </div>
    <hr />
    <p style={{
      textAlign: 'center', fontSize: '10pt', marginTop: '10px', marginBottom: 0, paddingBottom: '4px',
    }}
    >
      <i>
        Like Us On
        {' '}
        <a style={{ fontSize: '10pt' }} href="https://www.facebook.com/CollegeLutheranChurch/">Facebook</a>
      </i>
    </p>
    <div style={{ maxWidth: '300px', margin: 'auto', marginBottom: 0 }}>
      <iframe
        title="clc-facebook"/* eslint-disable-next-line max-len */
        src="//www.facebook.com/plugins/likebox.php?href=https%3A%2F%2Fwww.facebook.com%2FCollegeLutheranChurch&width=300&height=500&colorscheme=light&show_faces=false&header=true&stream=true&show_border=false"
        width="300"
        height="300"
        style={{ border: 'none', overflow: 'hidden' }}
        scrolling="no"
        frameBorder="0"
        allow="encrypted-media"
      />
    </div>
    <hr />
    <p style={{
      textAlign: 'center', fontSize: '10pt', marginTop: '10px', marginBottom: 0, paddingBottom: '4px',
    }}
    >
      {' '}
    </p>
    <div className="familySlideContainer">
      <div
        id="familySlideshowWide"
        style={{
          display: 'flex', flexDirection: 'column', marginTop: '1px', marginRight: 0,
        }}
      >
        {allPics && allPics.length > 0 ? (<PicSlider data={allPics} />) : null}
      </div>
    </div>
  </div>
);

FacebookFeed.defaultProps = { allPics: [] };

export default FacebookFeed;
