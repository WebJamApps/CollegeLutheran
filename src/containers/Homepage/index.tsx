import React, { RefObject } from 'react';
import { connect } from 'react-redux';
import { withResizeDetector } from 'react-resize-detector';
import About from './About';
import WideFacebookFeed from './WideFacebookFeed';
import NarrowFacebookFeed from './NarrowFacebookFeed';
import mapStoreToProps, { Ibook } from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

type HomepageProps = {
  targetRef: RefObject<HTMLDivElement>;
  width: number;
  height: number;
  homeContent?: Ibook;
  familyPics?: Ibook[];
  youthPics?: Ibook[];
  otherPics?: Ibook[];
  musicPics?: Ibook[];
};

interface HomepageState {
  picsState: Ibook[];
  homeContent?: Ibook;
}

export class Homepage extends React.Component<HomepageProps, HomepageState> {
  commonUtils: typeof commonUtils;

  parentRef: React.RefObject<unknown>;

  constructor(props: HomepageProps) {
    super(props);
    this.commonUtils = commonUtils;
    this.parentRef = React.createRef();
    this.state = { picsState: [] };
  }

  async componentDidMount(): Promise<void> {
    this.commonUtils.setTitleAndScroll('', window.screen.width);
    const delay = (): Promise<void> => new Promise((res) => setTimeout(res, 4000));
    return this.commonUtils.randomizePics(this, delay);
  }

  elca(w: number): JSX.Element { // eslint-disable-line class-methods-use-this
    const width = w < 420 ? '300px' : '400px';
    return (
      <div style={{
        textAlign: 'center', margin: 'auto', paddingTop: 0, paddingBottom: 0,
      }}
      >
        <a href="http://www.elca.org/" target="_blank" rel="noopener noreferrer">
          <img
            id="elcaLogo"
            alt="ELCA LOGO"
            src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
            style={{ width, margin: 'auto auto auto -2px', paddingTop: '30px' }}
          />
        </a>
        <p>{' '}</p>
      </div>
    );
  }

  render(): JSX.Element {
    const { picsState } = this.state;
    const { width, targetRef } = this.props;
    const { homeContent } = this.props;
    return (
      <div ref={targetRef}>
        {width >= 900
          ? (
            <div className="page-content">
              <About homeContent={homeContent} width={width} allPics={picsState} />
              <hr />
              <WideFacebookFeed width={width} />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
            </div>
          )
          : (
            <div className="page-content">
              <About homeContent={homeContent} width={width} allPics={picsState} />
              <hr />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
              <NarrowFacebookFeed allPics={picsState} />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
            </div>
          )}
        {this.elca(width)}
      </div>
    );
  }
}

export default connect(mapStoreToProps, null)(withResizeDetector(Homepage));
