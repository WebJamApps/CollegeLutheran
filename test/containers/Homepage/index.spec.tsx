/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { shallow } from 'enzyme';
import { Homepage } from '../../../src/containers/Homepage';
import WideAboutUs from '../../../src/containers/Homepage/About';
import WideFacebookFeed from '../../../src/containers/Homepage/WideFacebookFeed';
import NarrowFacebookFeed from '../../../src/containers/Homepage/NarrowFacebookFeed';
import PicSlider from '../../../src/components/PicSlider';

const targetRef:any = {};
const wrapper = shallow<Homepage>(<Homepage targetRef={targetRef} width={1000} height={800} />);

describe('Home', () => {
  it('Renders the homepage', () => {
    wrapper.instance().forceUpdate();
    expect(wrapper.find(WideAboutUs).exists()).toBe(true);
    expect(wrapper.find(WideAboutUs).dive().find('div.aboutPage').exists()).toBe(true);
    expect(wrapper.find(WideFacebookFeed).dive()
      .find('div')
      .exists()).toBe(true);
  });
  it('renders snapshot correctly', () => { expect(wrapper).toMatchSnapshot(); });
  it('Renders WideFacebook when at least 1092', () => {
    const wrapper2 = shallow(<WideFacebookFeed width={1092} />);
    expect(wrapper2.find('p#wideFacebook').prop('style')).toHaveProperty('marginBottom', '32px');
  });
  it('Renders NarrowFacebook with familyPics', () => {
    const allPics:any = {
      title: '', type: '', _id: '', created_at: '',
    };
    const wrapper2 = shallow(<NarrowFacebookFeed allPics={[allPics]} />);
    expect(wrapper2.find(PicSlider).exists()).toBe(true);
  });
  it('Renders WideAbout with familyPics', () => {
    const homeContent: any = '';
    const allPics:any[] = [{
      title: '', type: '', _id: '', created_at: '',
    }];
    const wrapper2 = shallow(<WideAboutUs
      allPics={allPics}
      homeContent={homeContent}
    />);
    expect(wrapper2.find(PicSlider).exists()).toBe(true);
  });
  it('renders the homepage with cellphone width', () => {
    const wrapper2 = shallow<Homepage>(<Homepage width={320} height={800} targetRef={targetRef} />);
    expect(wrapper2.find(NarrowFacebookFeed).exists()).toBe(true);
  });
});
