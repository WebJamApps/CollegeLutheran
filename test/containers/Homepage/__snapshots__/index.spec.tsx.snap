// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders snapshot correctly 1`] = `ShallowWrapper {}`;
